### PR TITLE
MGMT-17115: Save the custom catalog sources as extra manifests

### PR DESCRIPTION
**Pull request by eranco74**
LCA should preserve the user-defined catalog sources 
This change will fetch the catalog sources pre-reboot, filter out the default catalog sources, and add the user-defined catalog sources as as extra manifests that will get applied post-reboot (before the application restore starts)

### CODE REVIEW

File: internal/clusterconfig/clusterconfig.go [hunk 1]
```diff
@@ -427,3 +452,39 @@ func (r *UpgradeClusterConfigGather) fetchNetworkConfig(ostreeDir string) error
...
+
+func (r *UpgradeClusterConfigGather) getCustomCatalogSources(ctx context.Context) (operatorsv1alpha1.CatalogSourceList, error) {
+	catalogSources := &operatorsv1alpha1.CatalogSourceList{}
+	customCatalogSources := &operatorsv1alpha1.CatalogSourceList{}
+	allNamespaces := client.ListOptions{Namespace: metav1.NamespaceAll}
+	if err := r.Client.List(ctx, catalogSources, &allNamespaces); err != nil {
+		return operatorsv1alpha1.CatalogSourceList{}, fmt.Errorf("failed to list all catalogueSources %w", err)
+	}
+
+	defaultCatalogRecources := []string{"certified-operators", "community-operators", "redhat-marketplace", "redhat-operators"}
```
Comment (omertuc) [inline internal/clusterconfig/clusterconfig.go:464]: Maybe align with https://github.com/openshift-kni/lifecycle-agent/blob/e3bd855206b7ec61ec93822217fdd92158b23dc8/lca-cli/seedcreator/seedcreator.go#L478-L485

**Review by omertuc — CHANGES_REQUESTED** [review]

File: internal/clusterconfig/clusterconfig.go (hunk 1 above)
Comment (eranco74) [reply internal/clusterconfig/clusterconfig.go:464]: Hmm, I think that one is filtering the images, but let me check with @tsorya

Comment (tsorya) [reply internal/clusterconfig/clusterconfig.go:464]: yep, this regex matches image names

Comment (eranco74) [reply internal/clusterconfig/clusterconfig.go:464]: So I think for this case, we just want to filter out the default catalog sources and we achieve that by filtering the catalog source name.

**Comment by eranco74** [comment]
/hold
testing it...

**Comment by eranco74** [comment]
/unhold
The catalogsources looks good post reboot

**Pull request merged**